run_mount.go

```diff
@@ -41,6 +51,42 @@ func generateMountOpts(clicontext *cli.Context, imageVolumes map[string]struct{}
 		anonVolumes []string
 	)
 	mounted := make(map[string]struct{})
+	var imageVolumes map[string]struct{}
+	var tempDir string
+
+	if ensuredImage != nil {
+		imageVolumes = ensuredImage.ImageConfig.Volumes
+
+		if err := ensuredImage.Image.Unpack(ctx, clicontext.String("snapshotter")); err != nil {
+			return nil, nil, errors.Wrap(err, "error unpacking image")
+		}
+
+		diffIDs, err := ensuredImage.Image.RootFS(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+		chainID := identity.ChainID(diffIDs).String()
+
+		s := client.SnapshotService(clicontext.String("snapshotter"))
+		tempDir, err = ioutil.TempDir("", "initialC")
+		if err != nil {
+			return nil, nil, err
+		}
+		defer os.RemoveAll(tempDir)
+
+		var mounts []mount.Mount
+		mounts, err = s.View(ctx, tempDir, chainID)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		if err := mount.All(mounts, tempDir); err != nil {
```

Comment:
Where are the mounts getting cleaned up? The temp mount logic is ideal for these type of use cases

Reply:
@dmcgowan I added unmount logic .  The temp mount logic is  ideal but not in this context because it  push me to make same steps ( tmp dir creation then mount then unmount ) in each iteration `flagVSlice`

Reply:
I think it depends on `user cannot CONTRL-C during copy`. Need to figure out how to clean up in this case

Reply:
https://github.com/containerd/nerdctl/pull/216